common/test/src/ui/tst_MapDocument.cpp

```diff
@@ -462,6 +463,73 @@ TEST_CASE_METHOD(MapDocumentTest, "canUpdateLinkedGroups")
     document->canUpdateLinkedGroups(kdl::vec_static_cast<mdl::Node*>(entityNodes)));
 }
 
+TEST_CASE_METHOD(MapDocumentTest, "newDocument")
+{
+  const auto fgdContent = R"(
+    @SolidClass = worldspawn : "World entity" [
+      some_default_prop(string) : "" : "value"
+    ]
+  )";
+
+  constexpr auto configTemplate = R"(
+    {{
+        version: 9,
+        name: "testNewDocument",
+        icon: "Icon.png",
+        "fileformats": [ {{ "format": "Standard" }} ],
+        "filesystem": {{
+            "searchpath": ".",
+            "packageformat": {{ "extension": ".pak", "format": "idpak" }}
+        }},
+        "materials": {{
+            "root": "textures",
+            "format": {{ "extensions": [".jpg"], "format": "image" }}
+        }},
+        "entities": {{
+            "definitions": [ "WorldSpawnOnly.fgd" ],
+            "defaultcolor": "0.6 0.6 0.6 1.0",
+            "setDefaultProperties": {}
+        }},
+        "tags": {{ "brush": [], "brushface": [] }},
+        "faceattribs": {{ "surfaceflags": [], "contentflags": [] }}
+    }}
+  )";
+
+  SECTION("Default worldspawn properties")
+  {
+    using T = std::tuple<bool, std::vector<mdl::EntityProperty>>;
+    const auto& [setDefaultProperties, expectedEntityProperties] = GENERATE(values<T>({
+      {false,
+       {{mdl::EntityPropertyKeys::Classname,
+         mdl::EntityPropertyValues::WorldspawnClassname}}},
+      {true,
+       {{mdl::EntityPropertyKeys::Classname,
+         mdl::EntityPropertyValues::WorldspawnClassname},
+        {"some_default_prop", "value"}}},
+    }));
+
+    const auto configContent =
+      fmt::format(configTemplate, setDefaultProperties ? "true" : "false");
+
+    const auto gameName = "testNewDocument";
+    const auto dirPath = std::filesystem::current_path() / "fixture/games" / gameName;
+
+    auto environment = io::TestEnvironment{dirPath.string(), [&](auto& env) {
+                                     env.createFile("GameConfig.cfg", configContent);
+                                     env.createFile("WorldSpawnOnly.fgd", fgdContent);
+                                   }};
+
+    auto [gameTest, gameConfig] = mdl::loadGame(gameName);
+
+    document->newDocument(mdl::MapFormat::Standard, document->worldBounds(), gameTest)
```

Comment:
This doesn't work, and it crashes with ASAN enabled. You have to use `newMapDocument` from `TestUtils` to create a new document.

Reply:
Now it should work.